### PR TITLE
feat: Allow DeltaTable input to scan_delta and read_delta

### PR DESCRIPTION
**Pull request by deanm0000**
https://github.com/pola-rs/polars/issues/19074

### CODE REVIEW

File: py-polars/polars/io/delta.py [hunk 1]
```diff
@@ -314,12 +316,27 @@ def _get_delta_lake_table(
...
+    if isinstance(table_path, deltalake.DeltaTable):
+        if any(
+            [
+                version is not None,
+                storage_options is not None,
+                delta_table_options is not None,
+            ]
+        ):
+            warnings.warn(
+                """When supplying a DeltaTable directly, version, storage_options, and delta_table_options are ignored.
```
Comment (alexander-beedie) [inline py-polars/polars/io/delta.py:328]:
```suggestion
                """When supplying a DeltaTable directly, `version`, `storage_options`, and `delta_table_options` are ignored.
```

File: py-polars/polars/io/delta.py [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
+from polars import DataFrame
 from polars.convert import from_arrow
 from polars.datatypes import Null, Time
 from polars.datatypes.convert import unpack_dtypes
 from polars.dependencies import _DELTALAKE_AVAILABLE, deltalake
 from polars.io.pyarrow_dataset import scan_pyarrow_dataset
 
 if TYPE_CHECKING:
-    from polars import DataFrame, DataType, LazyFrame
+    from polars import DataType, LazyFrame
```
Comment (alexander-beedie) [inline py-polars/polars/io/delta.py:18]:
```suggestion
    from deltalake import DeltaTable

```

File: py-polars/polars/io/delta.py [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
 
 
 def read_delta(
-    source: str,
+    source: str | deltalake.DeltaTable,
```
Comment (alexander-beedie) [inline py-polars/polars/io/delta.py:21]:
```suggestion
    source: str | DeltaTable,
```

File: py-polars/polars/io/delta.py [hunk 4]
```diff
@@ -138,22 +140,23 @@ def read_delta(
...
+        DataFrame,
+        from_arrow(
+            dl_tbl.to_pyarrow_table(columns=columns, **pyarrow_options), rechunk=rechunk
+        ),
+    )
 
 
 def scan_delta(
-    source: str,
+    source: str | deltalake.DeltaTable,
```
Comment (alexander-beedie) [inline py-polars/polars/io/delta.py:159]:
```suggestion
    source: str | DeltaTable,
```

File: py-polars/polars/io/delta.py [hunk 5]
```diff
@@ -299,7 +301,7 @@ def _resolve_delta_lake_uri(table_uri: str, *, strict: bool = True) -> str:
 
 
 def _get_delta_lake_table(
-    table_path: str,
+    table_path: str | deltalake.DeltaTable,
```
Comment (alexander-beedie) [inline py-polars/polars/io/delta.py:304]:
```suggestion
    table_path: str | DeltaTable,
```

**Review by alexander-beedie — CHANGES_REQUESTED** [review]
A few minor suggestions (typing, error message), but otherwise I think allowing DeltaTable input here makes plenty of sense 👍

**Comment by deanm0000** [comment]
@alexander-beedie I accepted the tweaks but the code coverage doesn't like it. I'm not sure why, maybe it just needs a retry.

**Comment by alexander-beedie** [comment]
> @alexander-beedie I accepted the tweaks but the code coverage doesn't like it. I'm not sure why, maybe it just needs a retry.

Yup, those failures are unrelated.

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**